src/azul/debug.py

```diff
@@ -0,0 +1,67 @@
+from contextlib import (
+    ContextDecorator,
+)
+from logging import (
+    getLogger,
+)
+
+from azul import (
+    config,
+)
+
+log = getLogger(__name__)
+
+
+class RemoteDebugSession(ContextDecorator):
+    """
+    Adapted from https://stackoverflow.com/a/60333250/7830612
+
+    According to the Stackoverflow post, since the debug session is over a
+    long-running TCP connection, we need to close the connection explicitly
+    within the lambda, otherwise the lambda will time out. A context manager
+    handles this conveniently.
+
+    For usage instructions, see the README.
+    """
+
+    def __init__(self):
+        self.active = False
+        if self.is_available():
+            import pydevd
+            self.pydevd = pydevd
+        else:
+            self.pydevd = None
+            log.warning('Remote debugging is not available')
+
+    def __enter__(self):
+        if self.is_available():
+            self.pydevd.settrace(config.debug_host,
+                                 port=config.debug_port,
+                                 suspend=False,
+                                 stdoutToServer=True,
+                                 stderrToServer=True)
+            log.info('Starting remote debugging session')
+            self.active = True
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        if self.active:
+            log.info(f'Stopping remote debugging on {config.debug_host}:{config.debug_port}')
+            self.pydevd.stoptrace()
+            self.active = False
+        return False
+
+    @classmethod
+    def is_available(cls):
+        try:
+            import pydevd  # noqa: F401
+        except ImportError:
+            available = False
```

Comment:
```suggestion
            return False
```

Reply:
This is no longer relevant given some other refactoring I've done.